components/apimgt/org.wso2.carbon.apimgt.rest.api.commons/src/main/java/org/wso2/carbon/apimgt/rest/api/common/impl/OAuth2Authenticator.java

```diff
@@ -148,24 +148,30 @@ private AccessTokenInfo validateToken(String accessToken) throws APIMgtSecurityE
      * @throws APIMgtSecurityException if the Authorization header is invalid
      */
     private String extractPartialAccessTokenFromCookie(String cookie) {
-        cookie = cookie.trim();
-        String[] cookies = cookie.split(";");
-        String token2 = Arrays.stream(cookies)
-                .filter(name -> name.contains(APIConstants.AccessTokenConstants.AM_TOKEN_MSF4J))
-                .findFirst().orElse("");
-        if (token2.split("=").length == 2) {
-            return token2.split("=")[1];
+        if (cookie != null) {
+            cookie = cookie.trim();
+            String[] cookies = cookie.split(";");
+
+            String token2 = Arrays.stream(cookies)
+                    .filter(name -> name.contains(APIConstants.AccessTokenConstants.AM_TOKEN_MSF4J))
+                    .findFirst().orElse("");
+            if (token2.split("=").length == 2) {
```

Comment:
Let's assign token2.split("=") to a variable so we can use it in the next line (instead of splitting again.)